var/spack/repos/builtin/packages/phist/package.py

```diff
@@ -137,6 +121,12 @@ class Phist(CMakePackage):
         description="generate Fortran 2003 bindings (requires Python3 and " "a Fortran compiler)",
     )
 
+    # ###################### Variant conflicts ##########################
+
+    # spack GitLab CI pipelines use ^mpich %gcc@7.5.0, but @1.9.6: kernel_lib=builtin fails
+    # due to the use of the mpi_f08 module in phist/builtin. We therefore force the concretizer
+    # to pick another kernel_lib (e.g. epetra or petsc) by this conflict statement.
+    conflicts("kernel_lib=builtin", when="@1.9.6: ^mpich %gcc@:10")
```

Comment:
Directives should be used in `package.py` if they are "universally" true. It seems from the comment that this one is needed only to drive the concretization of pipelines. In that case the modifications should go in the corresponding `spack.yaml` file.